app/services/GuidanceService.scala

```diff
@@ -45,6 +45,23 @@ class GuidanceService @Inject() (
 
   val logger: Logger = Logger(getClass)
 
+  def sessionRestart(processCode: String, sessionId: String)(implicit context: ExecutionContext): Future[RequestOutcome[String]] =
+    sessionRepository.getResetSession(sessionId).map{
+      case Right(ctx) if processCode == ctx.process.meta.processCode =>
+        (ctx.urlToPageId.toList.map(e => (e._2, e._1)).toMap)
```

Comment:
You could use map.find(_._2 = ctx.process.startPageId) which would give you Some((url,id)). Perhaps simpler.

Reply:
done